### Carregamento de bibliotecas

No bloco de código abaixo é feito o carregamento de todas as bibliotecas que serão utilizadas neste notebook.

In [57]:
from scipy.io import loadmat
import numpy as np

### Pré-carregamento

No bloco de código abaixo, realizamos o carregamento de todos os predaçõs da base de dados avi em `single-channel`. 

É efetuado uma iteração entre 1 e 5 e concatenado os dados carregados na variavel `avi_data`.

In [58]:
# pré-carregamento

avi_data = list()
for i in range(1,5):
    avi_data.append(loadmat(f"datasets/avi/single/Sub{i}_singletarget.mat"))

### Captura dos Dados

No trecho de código abaixo é executado a captura dos dados (X) e suas respectivas classes (y). 

In [64]:
# extração direta dos dados

X = [d['Data'][0][0][0] for d in avi_data]
y = [d['Data'][0][0][1] for d in avi_data]

print("Divisão dos dados:")
for data in X:
    print(data.shape)

print("\nDivisão dos rótulos sem duplicatas (sequências):")
for data in y:
    print(data)

Divisão dos dados:
(15360, 27)
(15360, 26)
(15360, 21)
(15360, 21)

Divisão dos rótulos sem duplicatas (sequências):
[[10.  10.  10.  12.  12.  12.   6.5  6.5  6.5  6.   6.   6.   6.   6.
   6.   7.5  7.5  7.5  7.   7.   7.   8.2  8.2  8.2  9.3  9.3  9.3]]
[[10.  10.  10.   6.5  6.5  6.5  6.5  6.   6.   6.   7.5  7.5  7.5  7.
   7.   7.   7.   7.   7.   8.2  8.2  8.2  8.2  9.3  9.3  9.3]]
[[10.  10.  10.   6.5  6.5  6.5  6.   6.   6.   7.5  7.5  7.5  7.   7.
   7.   8.2  8.2  8.2  9.3  9.3  9.3]]
[[10.  10.  10.   6.5  6.5  6.5  6.   6.   6.   7.5  7.5  7.5  7.   7.
   7.   8.2  8.2  8.2  9.3  9.3  9.3]]


### Padronização

No trecho de código abaixo removemos frequências duplicadas que estão presentes nos dados do experimento.

Essas frequências podem ser entendidas como "lixo".

In [60]:
# padronizando 1º participante
X[0] = X[0][:, list(set(range(27)) - set([3, 4, 5, 12, 13, 14]))]
y[0] = y[0][:, list(set(range(27)) - set([3, 4, 5, 12, 13, 14]))]

# padronizando 2º participante
X[1] = X[1][:, list(set(range(26)) - set([6, 16, 17, 18, 22]))]
y[1] = y[1][:, list(set(range(26)) - set([6, 16, 17, 18, 22]))]

print("Divisão dos dados:")
for data in X:
    print(data.shape)

print("\nDivisão dos rótulos (sequências):")
for data in y:
    print(data)


Divisão dos dados:
(15360, 21)
(15360, 21)
(15360, 21)
(15360, 21)

Divisão dos rótulos (sequências):
[[10.  10.  10.   6.5  6.5  6.5  6.   6.   6.   7.5  7.5  7.5  7.   7.
   7.   8.2  8.2  8.2  9.3  9.3  9.3]]
[[10.  10.  10.   6.5  6.5  6.5  6.   6.   6.   7.5  7.5  7.5  7.   7.
   7.   8.2  8.2  8.2  9.3  9.3  9.3]]
[[10.  10.  10.   6.5  6.5  6.5  6.   6.   6.   7.5  7.5  7.5  7.   7.
   7.   8.2  8.2  8.2  9.3  9.3  9.3]]
[[10.  10.  10.   6.5  6.5  6.5  6.   6.   6.   7.5  7.5  7.5  7.   7.
   7.   8.2  8.2  8.2  9.3  9.3  9.3]]


### Organização dos dados

No trecho de código abaixo é realizado o `reshape` dos dados presentes na variável `X`.

O algoritmo realizará a seguinte transformação:

De

- (4, 1560, 21)

Para

- (4, 21, 15360)

In [63]:
reshape_X = [np.transpose(data) for data in X]
for data in reshape_X:
    print(data.shape)


(21, 15360)
(21, 15360)
(21, 15360)
(21, 15360)


### Salvando dados em um arquivo

No trecho de código abaixo é feito o salvamento dos dados formatados na extensão `.npy`.

In [62]:
# salva os dados numpy

np.save("datasets/avi/single/data_sub1.npy", reshape_X)
np.save("datasets/avi/single/labels_sub1.npy", y)

### Tarefas:

- Padronizar os dados de todos os 4 participantes dos dados *single-target* e salvar de forma adequada para utilização nos próximos notebooks.
- Realizar o mesmo estudo para os dados *multi-target*